platform-workflow-module/src/main/java/com/softicar/platform/workflow/module/workflow/transition/WorkflowTransitionValidator.java

```diff
@@ -102,9 +101,15 @@ private void assertValidSideEffect(ISourceCodeReferencePoint sideEffect) {
 
 	private void assertValidSideEffect(IWorkflowTransitionSideEffect<?> sideEffect) {
 
-		IEmfTable<?, ?, ?> table = tableRow.getWorkflowVersion().getWorkflow().getTableReferencePointOrThrow().getTable();
-		if (!sideEffect.getValueClass().isAssignableFrom(table.getValueClass())) {
-			addError(AGWorkflowTransition.SIDE_EFFECT, WorkflowI18n.SIDE_EFFECT_IS_NOT_COMPATIBLE_WITH_ARG1.toDisplay(table.getTitle()));
-		}
+		tableRow//
+			.getWorkflowVersion()
+			.getWorkflow()
+			.getTableReferencePoint()
+			.ifPresent(referencePoint -> {
+				var table = referencePoint.getTable();
+				if (!sideEffect.getValueClass().isAssignableFrom(table.getValueClass())) {
```

Comment:
We could extract the content of the `if` parentheses to a method, e.g. `isCompatible(sideEffect, table)` ...or `isIncompatible`